### PR TITLE
feat: update to latest mongosh, driver, and segment analytics VSCODE-366

### PR DESCRIPTION
**Pull request by alenakhineika**
## Description

Update to:
- mongosh@^1.8.0
- mongodb@^5.1.0
- bson@^5.1.0
- mongodb-schema@^10.0.1
- analytics-node@^6.2.0
- vscode engines @ ^1.76.2

### Checklist
- [ ] New tests and/or benchmarks are included
- [ ] Documentation is changed or added
- [ ] I have signed the MongoDB Contributor License Agreement (https://www.mongodb.com/legal/contributor-agreement)

## Motivation and Context

- [ ] Bugfix
- [ ] New feature
- [x] Dependency update
- [ ] Misc

## Types of changes

- [ ] *Backport Needed*
- [x] Patch (non-breaking change which fixes an issue)
- [ ] Minor (non-breaking change which adds functionality)
- [ ] Major (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: src/editors/mongoDBDocumentService.ts [hunk 1]
```diff
@@ -90,8 +94,8 @@ export default class MongoDBDocumentService {
       );
       await findOneAndReplace(
         namespace,
-        { _id: documentId },
-        newDocument as Document,
+        { _id: documentId as ObjectId },
```
Comment (Anemy) [inline src/editors/mongoDBDocumentService.ts:97]: Do we need to cast this to ObjectId here? Maybe there's a better type like `Document` or `any`, as _id can be types besides `ObjectId`.
```suggestion
        { _id: documentId as ObjectId },
```

File: src/editors/mongoDBDocumentService.ts [hunk 2]
```diff
@@ -135,19 +137,17 @@ export default class MongoDBDocumentService {
     try {
       const documents = await dataservice.find(
         namespace,
-        { _id: documentId },
+        { _id: documentId as ObjectId },
```
Comment (Anemy) [inline src/editors/mongoDBDocumentService.ts:140]: Same here as above

File: src/explorer/documentTreeItem.ts [hunk 3]
```diff
@@ -61,7 +62,7 @@ export default class DocumentTreeItem
   async getDocumentContents(): Promise<Document> {
     const documents = await this.dataService.find(
       this.namespace,
-      { _id: this.documentId },
+      { _id: this.documentId as ObjectId },
```
Comment (Anemy) [inline src/explorer/documentTreeItem.ts:65]: Also here and below in this file

**Review by Anemy — COMMENTED** [review]
Nice, left a question and a comment on _id types that might not be `ObjectId`s.

File: src/editors/mongoDBDocumentService.ts (hunk 1 above)
Comment (alenakhineika) [reply src/editors/mongoDBDocumentService.ts:97]: Maybe any would be better 👍 Document doesn't seem fit for string values.

Comment (Anemy) [reply src/editors/mongoDBDocumentService.ts:97]: Yeah scratch that `Document` suggestion, it's not the right type here. any is fine by me 👍

File: webpack.config.js [hunk 4]
```diff
@@ -16,6 +16,9 @@ const baseConfig = {
 
 const extensionConfig = {
   ...baseConfig,
+  experiments: {
+    topLevelAwait: true,
+  },
```
[the remark this answers is not in the document]
Comment (alenakhineika) [reply webpack.config.js:21]: Something changed in the `bson` library I believe that required a top-level await, so the extension crashed on startup with the error:
```log
2023-03-20 18:06:02.608 [error] Activating extension mongodb.mongodb-vscode failed due to an error:
2023-03-20 18:06:02.608 [error] Error: Module parse failed: The top-level-await experiment is not enabled (set experiments.topLevelAwait: true to enabled it)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
Error: The top-level-await experiment is not enabled (set experiments.topLevelAwait: true to enabled it)
    at /Users/alena.khineika/www/vscode/node_modules/webpack/lib/dependencies/HarmonyDetectionParserPlugin.js:54:11
    at Hook.eval [as call] (eval at create (/Users/alena.khineika/www/vscode/node_modules/tapable/lib/HookCodeFactory.js:19:10), <anonymous>:7:16)
...
    at ./node_modules/bson/lib/bson.mjs (/Users/alena.khineika/www/vscode/dist/extension.js:90406:7)
    at __webpack_require__ (/Users/alena.khineika/www/vscode/dist/extension.js:336777:43)
```

**Review by Anemy — APPROVED** [review]

**Pull request merged**